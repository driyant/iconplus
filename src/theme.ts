// src/theme.ts
import { extendTheme } from '@chakra-ui/react';
import '@fontsource/open-sans/400.css';
import '@fontsource/open-sans/700.css';

const theme = extendTheme({
  initialColorMode: 'light',
  useSystemColorMode: false,
  fonts: {
    heading: 'Open Sans, sans-serif',
    body: 'Open Sans, sans-serif',
  },
  components: {
    // Tambahkan custom components jika diperlukan
  },
});

export default theme;
