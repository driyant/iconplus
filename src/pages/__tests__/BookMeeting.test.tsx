import { describe, it, expect, vi, beforeEach } from 'vitest';
import { ChakraProvider } from '@chakra-ui/react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import BookMeeting from '../BookMeeting';
import theme from '../../theme';

vi.mock('@fontsource/open-sans/400.css', () => ({}));
vi.mock('@fontsource/open-sans/700.css', () => ({}));
vi.mock('@chakra-ui/react', async (importOriginal) => {
  const actual = await importOriginal<typeof import('@chakra-ui/react')>();
  return {
    ...actual,
    useToast: () => vi.fn(),
    extendTheme: actual.extendTheme,
  };
});

const mockFetch = vi.fn();
global.fetch = mockFetch;
vi.spyOn(console, 'error').mockImplementation(() => {});

const renderBookMeeting = () => {
  return render(
    <ChakraProvider theme={theme} resetCSS>
      <BookMeeting />
    </ChakraProvider>
  );
};

describe('BookMeeting Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
    mockFetch.mockImplementation((url) => {
      if (url.includes('masterOffice') || url.includes('/mockApi/masterOffice.json')) {
        return Promise.resolve({
          ok: true,
          json: () =>
            Promise.resolve([
              { id: '1', officeName: 'UID BALI' },
              { id: '2', officeName: 'UID JAKARTA' },
            ]),
        });
      }
      if (url.includes('masterMeetingRooms') || url.includes('/mockApi/masterMeetingRooms.json')) {
        return Promise.resolve({
          ok: true,
          json: () =>
            Promise.resolve([
              { id: '101', roomName: 'Ruang Ganesha', officeId: '1', capacity: 10 },
            ]),
        });
      }
      if (
        url.includes('masterJenisKonsumsi') ||
        url.includes('/mockApi/masterJenisKonsumsi.json')
      ) {
        return Promise.resolve({
          ok: true,
          json: () =>
            Promise.resolve([
              { name: 'Snack Siang', maxPrice: 10000 },
              { name: 'Makan Siang', maxPrice: 20000 },
              { name: 'Snack Sore', maxPrice: 15000 },
            ]),
        });
      }
      return Promise.resolve({
        ok: false,
        status: 404,
        json: () => Promise.resolve({ error: 'Not found' }),
      });
    });
  });

  afterAll(() => {
    vi.restoreAllMocks();
  });

  it('should render without crashing', async () => {
    renderBookMeeting();
    expect(screen.getByText(/Informasi Ruang Meeting/i)).toBeInTheDocument();
  });

  it('should handle API fetch errors', async () => {
    mockFetch.mockRejectedValueOnce(new Error('Network error')); // mock Error masterOffice
    mockFetch.mockRejectedValueOnce(new Error('Network error')); // mock Error masterMeetingRooms
    mockFetch.mockRejectedValueOnce(new Error('Network error')); // mock Error masterJenisKonsumsi
    renderBookMeeting();
    await waitFor(() => {
      expect(console.error).toHaveBeenCalled();
    });
  });

  it('should load and display initial data input unit', async () => {
    const response = await mockFetch('/api/masterOffice');
    const data = await response.json();
    expect(response.ok).toBe(true);
    expect(data).toEqual([
      { id: '1', officeName: 'UID BALI' },
      { id: '2', officeName: 'UID JAKARTA' },
    ]);
  });

  it('should load and display initial data input room', async () => {
    const response = await mockFetch('/api/masterMeetingRooms');
    const data = await response.json();
    expect(response.ok).toBe(true);
    expect(data).toEqual([{ id: '101', roomName: 'Ruang Ganesha', officeId: '1', capacity: 10 }]);
  });

  it('should load and display initial data input meal', async () => {
    const response = await mockFetch('/api/masterJenisKonsumsi');
    const data = await response.json();
    expect(response.ok).toBe(true);
    expect(data).toEqual([
      { name: 'Snack Siang', maxPrice: 10000 },
      { name: 'Makan Siang', maxPrice: 20000 },
      { name: 'Snack Sore', maxPrice: 15000 },
    ]);
  });

  it('option pilih unit should have length > 1', async () => {
    const { getByTestId } = renderBookMeeting();
    await waitFor(() => {
      const unitSelect = getByTestId('unit-select') as HTMLSelectElement;
      expect(unitSelect.options.length).toBeGreaterThan(1);
      expect(unitSelect.options[1].textContent).toBe('UID BALI');
      expect(unitSelect.options[2].textContent).toBe('UID JAKARTA');
    });
  });

  it('should filter rooms when unit is selected', async () => {
    renderBookMeeting();

    await waitFor(() => {
      expect(screen.getByText('UID BALI')).toBeInTheDocument();
    });

    // Select unit
    fireEvent.change(screen.getByTestId('unit-select'), {
      target: { value: '1' },
    });

    await waitFor(() => {
      expect(screen.getByText('Ruang Ganesha')).toBeInTheDocument();
    });
  });

  it('should update capacity when room is selected', async () => {
    renderBookMeeting();

    // Select unit first
    await waitFor(() => {
      expect(screen.getByText('UID BALI')).toBeInTheDocument();
    });

    fireEvent.change(screen.getByTestId('unit-select'), {
      target: { value: '1' },
    });

    // select room
    await waitFor(() => {
      expect(screen.getByText('Ruang Ganesha')).toBeInTheDocument();
    });
    fireEvent.change(screen.getByLabelText(/Ruang Meeting/i), {
      target: { value: 'Ruang Ganesha' },
    });

    expect(screen.getByLabelText(/Kapasitas/i)).toHaveValue('10');
  });

  it('should calculate total price correctly', async () => {
    renderBookMeeting();

    // Load data
    await waitFor(() => {
      expect(screen.getByText('UID BALI')).toBeInTheDocument();
    });

    // Fill form
    fireEvent.change(screen.getByTestId('unit-select'), { target: { value: '1' } });
    fireEvent.change(screen.getByLabelText(/Ruang Meeting/i), {
      target: { value: 'Ruang Ganesha' },
    });
    fireEvent.change(screen.getByLabelText(/Jumlah Peserta/i), {
      target: { value: '5' },
    });
    // Select meals
    fireEvent.click(screen.getByLabelText(/Snack Siang/i));
    fireEvent.click(screen.getByLabelText(/Makan Siang/i));

    await waitFor(() => {
      expect(screen.getByLabelText(/Nominal Konsumsi/i)).toHaveValue('150.000');
    });
  });

  it('should submit the form and cover handleSubmitForm', async () => {
    renderBookMeeting();

    // Wait data
    await waitFor(() => {
      expect(screen.getByRole('option', { name: 'UID BALI' })).toBeInTheDocument();
    });

    // Unit
    fireEvent.change(screen.getByTestId('unit-select'), { target: { value: '1' } });

    // Wait for room meeting
    await waitFor(() => {
      expect(screen.getByRole('option', { name: 'Ruang Ganesha' })).toBeInTheDocument();
    });

    // Choose Room Meeting
    fireEvent.change(screen.getByLabelText(/Ruang Meeting/i), {
      target: { value: 'Ruang Ganesha' },
    });

    // Date
    fireEvent.change(screen.getByLabelText(/Tanggal Rapat/i), { target: { value: '2025-12-31' } });

    // Time start
    fireEvent.change(screen.getByLabelText(/Waktu Mulai/i), { target: { value: '10:00' } });

    // Time end
    await waitFor(() => {
      expect(
        screen.getByLabelText(/Waktu Selesai/i).querySelectorAll('option').length
      ).toBeGreaterThan(1);
    });

    // Pilih Waktu Selesai
    fireEvent.change(screen.getByLabelText(/Waktu Selesai/i), { target: { value: '12:00' } });

    // Capacity
    fireEvent.change(screen.getByLabelText(/Jumlah Peserta/i), { target: { value: '5' } });

    // Choose meals
    fireEvent.click(screen.getByLabelText(/Snack Siang/i));

    await waitFor(() => {
      expect(screen.getByRole('button', { name: /Simpan/i })).toBeEnabled();
    });

    // Submit form
    fireEvent.click(screen.getByRole('button', { name: /Simpan/i }));

    await waitFor(() => {
      // Reset form
      expect(screen.getByTestId('unit-select')).toHaveValue('');
    });
  });

  it('should auto-select meals based on timeStart and timeEnd', async () => {
    renderBookMeeting();

    await waitFor(() => {
      expect(screen.getByRole('option', { name: 'UID BALI' })).toBeInTheDocument();
    });
    fireEvent.change(screen.getByTestId('unit-select'), { target: { value: '1' } });
    await waitFor(() => {
      expect(screen.getByRole('option', { name: 'Ruang Ganesha' })).toBeInTheDocument();
    });
    fireEvent.change(screen.getByLabelText(/Ruang Meeting/i), {
      target: { value: 'Ruang Ganesha' },
    });

    // Only Snack Siang meeting: (start 09:00, end 10:00) ---
    fireEvent.change(screen.getByLabelText(/Waktu Mulai/i), { target: { value: '09:00' } });
    fireEvent.change(screen.getByLabelText(/Waktu Selesai/i), { target: { value: '10:00' } });
    await waitFor(() => {
      expect(screen.getByLabelText(/Snack Siang/i)).toBeChecked();
      expect(screen.getByLabelText(/Makan Siang/i)).not.toBeChecked();
      expect(screen.getByLabelText(/Snack Sore/i)).not.toBeChecked();
    });

    // Lunch + Afternoon break snack (start 10:00, end 12:00) ---
    fireEvent.change(screen.getByLabelText(/Waktu Mulai/i), { target: { value: '10:00' } });
    fireEvent.change(screen.getByLabelText(/Waktu Selesai/i), { target: { value: '12:00' } });
    await waitFor(() => {
      expect(screen.getByLabelText(/Snack Siang/i)).toBeChecked();
      expect(screen.getByLabelText(/Makan Siang/i)).toBeChecked();
      expect(screen.getByLabelText(/Snack Sore/i)).not.toBeChecked();
    });

    // Lunch meeting between (start 11:00, end 14:00) ---
    fireEvent.change(screen.getByLabelText(/Waktu Mulai/i), { target: { value: '11:00' } });
    fireEvent.change(screen.getByLabelText(/Waktu Selesai/i), { target: { value: '14:00' } });
    await waitFor(() => {
      expect(screen.getByLabelText(/Snack Siang/i)).not.toBeChecked();
      expect(screen.getByLabelText(/Makan Siang/i)).toBeChecked();
      expect(screen.getByLabelText(/Snack Sore/i)).not.toBeChecked();
    });

    // Lunch + Snack Sore meeting: (start: 14:00, end 17:00) ---
    fireEvent.change(screen.getByLabelText(/Waktu Mulai/i), { target: { value: '14:00' } });
    fireEvent.change(screen.getByLabelText(/Waktu Selesai/i), { target: { value: '17:00' } });
    await waitFor(() => {
      expect(screen.getByLabelText(/Snack Siang/i)).not.toBeChecked();
      expect(screen.getByLabelText(/Makan Siang/i)).toBeChecked();
      expect(screen.getByLabelText(/Snack Sore/i)).toBeChecked();
    });

    // Only Snack Sore (start 15:00, end 17:00) ---
    fireEvent.change(screen.getByLabelText(/Waktu Mulai/i), { target: { value: '15:00' } });
    fireEvent.change(screen.getByLabelText(/Waktu Selesai/i), { target: { value: '17:00' } });
    await waitFor(() => {
      expect(screen.getByLabelText(/Snack Siang/i)).not.toBeChecked();
      expect(screen.getByLabelText(/Makan Siang/i)).not.toBeChecked();
      expect(screen.getByLabelText(/Snack Sore/i)).toBeChecked();
    });

    // All meals meeting: (start 09:00, end 17:00) ---
    fireEvent.change(screen.getByLabelText(/Waktu Mulai/i), { target: { value: '09:00' } });
    fireEvent.change(screen.getByLabelText(/Waktu Selesai/i), { target: { value: '17:00' } });
    await waitFor(() => {
      expect(screen.getByLabelText(/Snack Siang/i)).toBeChecked();
      expect(screen.getByLabelText(/Makan Siang/i)).toBeChecked();
      expect(screen.getByLabelText(/Snack Sore/i)).toBeChecked();
    });
  });

  it('should allow manual check and uncheck Snack Sore checkbox', async () => {
    renderBookMeeting();

    await waitFor(() => {
      expect(screen.getByRole('option', { name: 'UID BALI' })).toBeInTheDocument();
    });
    fireEvent.change(screen.getByTestId('unit-select'), { target: { value: '1' } });
    await waitFor(() => {
      expect(screen.getByRole('option', { name: 'Ruang Ganesha' })).toBeInTheDocument();
    });
    fireEvent.change(screen.getByLabelText(/Ruang Meeting/i), {
      target: { value: 'Ruang Ganesha' },
    });

    expect(screen.getByLabelText(/Snack Sore/i)).not.toBeChecked();

    fireEvent.click(screen.getByLabelText(/Snack Sore/i));
    expect(screen.getByLabelText(/Snack Sore/i)).toBeChecked();

    fireEvent.click(screen.getByLabelText(/Snack Sore/i));
    expect(screen.getByLabelText(/Snack Sore/i)).not.toBeChecked();
  });
});
