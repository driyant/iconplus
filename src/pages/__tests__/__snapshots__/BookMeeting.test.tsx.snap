// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookMeeting Component > should create a snapshot after data loaded 1`] = `
<DocumentFragment>
  <div
    class="css-s9jvxl"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-kxotqw"
      >
        Ruang Meeting
      </p>
      <nav
        aria-label="breadcrumb"
        class="chakra-breadcrumb css-jhncyh"
      >
        <ol
          class="chakra-breadcrumb__list css-70qvj9"
        >
          <li
            class="chakra-breadcrumb__list-item css-18biwo"
          >
            <a
              class="chakra-breadcrumb__link css-1u2cnya"
            >
              Ruang Meeting
            </a>
            <span
              class="css-t4q1nq"
              role="presentation"
            >
              /
            </span>
          </li>
          <li
            class="chakra-breadcrumb__list-item css-18biwo"
          >
            <a
              class="chakra-breadcrumb__link css-1u2cnya"
            >
              Pesan Ruangan
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
  <div
    class="css-1cuepr6"
  >
    <p
      class="chakra-text css-1d281jg"
    >
      Informasi Ruang Meeting
    </p>
    <form
      class="css-0"
    >
      <div
        class="css-1f3mncv"
      >
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-g6pte"
            for="field-«r0»"
            id="field-«r0»-label"
          >
            Unit
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-1tfjd1n"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              aria-required="true"
              class="chakra-select css-161pkch"
              data-testid="unit-select"
              id="field-«r0»"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Pilih Unit
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-iohxn1"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentcolor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-g6pte"
            for="field-«r1»"
            id="field-«r1»-label"
          >
            Ruang Meeting
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-1tfjd1n"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              aria-required="true"
              class="chakra-select css-161pkch"
              id="field-«r1»"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Pilih Ruang Meeting
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-iohxn1"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentcolor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1f3mncv"
      >
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-g6pte"
            for="field-«r2»"
            id="field-«r2»-label"
          >
            Kapasitas
          </label>
          <input
            aria-readonly="true"
            class="chakra-input css-1pope8p"
            id="field-«r2»"
            readonly=""
            value="0"
          />
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-rnrgf4"
      />
      <p
        class="chakra-text css-1d281jg"
      >
        Informasi Rapat
      </p>
      <div
        class="css-1m7uxgx"
      >
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-g6pte"
            for="field-«r3»"
            id="field-«r3»-label"
          >
            Tanggal Rapat
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-1tfjd1n"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-input__group css-1y0e7gb"
            data-group="true"
          >
            <div
              class="chakra-input__left-addon css-127xmqf"
            >
              <svg
                class="chakra-icon css-13otjrl"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="18"
                  rx="2"
                  ry="2"
                  width="18"
                  x="3"
                  y="4"
                />
                <line
                  x1="16"
                  x2="16"
                  y1="2"
                  y2="6"
                />
                <line
                  x1="8"
                  x2="8"
                  y1="2"
                  y2="6"
                />
                <line
                  x1="3"
                  x2="21"
                  y1="10"
                  y2="10"
                />
              </svg>
            </div>
            <input
              aria-required="true"
              class="chakra-input css-10wjqo2"
              id="field-«r3»"
              min="2025-06-14"
              placeholder="28 Juni 2022"
              required=""
              type="date"
              value=""
            />
          </div>
        </div>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-g6pte"
            for="field-«r4»"
            id="field-«r4»-label"
          >
            Waktu Mulai
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-1tfjd1n"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              aria-required="true"
              class="chakra-select css-161pkch"
              id="field-«r4»"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Pilih Waktu Mulai
              </option>
              <option
                value="01:00"
              >
                01:00
              </option>
              <option
                value="02:00"
              >
                02:00
              </option>
              <option
                value="03:00"
              >
                03:00
              </option>
              <option
                value="04:00"
              >
                04:00
              </option>
              <option
                value="05:00"
              >
                05:00
              </option>
              <option
                value="06:00"
              >
                06:00
              </option>
              <option
                value="07:00"
              >
                07:00
              </option>
              <option
                value="08:00"
              >
                08:00
              </option>
              <option
                value="09:00"
              >
                09:00
              </option>
              <option
                value="10:00"
              >
                10:00
              </option>
              <option
                value="11:00"
              >
                11:00
              </option>
              <option
                value="12:00"
              >
                12:00
              </option>
              <option
                value="13:00"
              >
                13:00
              </option>
              <option
                value="14:00"
              >
                14:00
              </option>
              <option
                value="15:00"
              >
                15:00
              </option>
              <option
                value="16:00"
              >
                16:00
              </option>
              <option
                value="17:00"
              >
                17:00
              </option>
              <option
                value="18:00"
              >
                18:00
              </option>
              <option
                value="19:00"
              >
                19:00
              </option>
              <option
                value="20:00"
              >
                20:00
              </option>
              <option
                value="21:00"
              >
                21:00
              </option>
              <option
                value="22:00"
              >
                22:00
              </option>
              <option
                value="23:00"
              >
                23:00
              </option>
              <option
                value="00:00"
              >
                00:00
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-iohxn1"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentcolor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-g6pte"
            for="field-«r5»"
            id="field-«r5»-label"
          >
            Waktu Selesai
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-1tfjd1n"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              aria-required="true"
              class="chakra-select css-161pkch"
              id="field-«r5»"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Pilih Waktu Selesai
              </option>
              <option
                value="01:00"
              >
                01:00
              </option>
              <option
                value="02:00"
              >
                02:00
              </option>
              <option
                value="03:00"
              >
                03:00
              </option>
              <option
                value="04:00"
              >
                04:00
              </option>
              <option
                value="05:00"
              >
                05:00
              </option>
              <option
                value="06:00"
              >
                06:00
              </option>
              <option
                value="07:00"
              >
                07:00
              </option>
              <option
                value="08:00"
              >
                08:00
              </option>
              <option
                value="09:00"
              >
                09:00
              </option>
              <option
                value="10:00"
              >
                10:00
              </option>
              <option
                value="11:00"
              >
                11:00
              </option>
              <option
                value="12:00"
              >
                12:00
              </option>
              <option
                value="13:00"
              >
                13:00
              </option>
              <option
                value="14:00"
              >
                14:00
              </option>
              <option
                value="15:00"
              >
                15:00
              </option>
              <option
                value="16:00"
              >
                16:00
              </option>
              <option
                value="17:00"
              >
                17:00
              </option>
              <option
                value="18:00"
              >
                18:00
              </option>
              <option
                value="19:00"
              >
                19:00
              </option>
              <option
                value="20:00"
              >
                20:00
              </option>
              <option
                value="21:00"
              >
                21:00
              </option>
              <option
                value="22:00"
              >
                22:00
              </option>
              <option
                value="23:00"
              >
                23:00
              </option>
              <option
                value="00:00"
              >
                00:00
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-iohxn1"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentcolor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1f3mncv"
      >
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-g6pte"
            for="field-«r6»"
            id="field-«r6»-label"
          >
            Jumlah Peserta
          </label>
          <input
            class="chakra-input css-1cjy4zv"
            id="field-«r6»"
            inputmode="numeric"
            pattern="[0-9]*"
            placeholder="Masukkan Jumlah Peserta"
            value=""
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-1n1vgb"
        role="group"
      >
        <label
          class="chakra-form__label css-g6pte"
          for="field-«r7»"
          id="field-«r7»-label"
        >
          Jenis Konsumsi
        </label>
        <div
          class="chakra-stack css-8g8ihq"
        >
          <label
            class="chakra-checkbox css-1577qb8"
          >
            <input
              aria-checked="false"
              aria-disabled="false"
              aria-invalid="false"
              class="chakra-checkbox__input"
              id="field-«r7»"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-1dnp747"
            />
            <span
              class="chakra-checkbox__label css-6x44c9"
            >
              Snack Siang
            </span>
          </label>
          <label
            class="chakra-checkbox css-1577qb8"
          >
            <input
              aria-checked="false"
              aria-disabled="false"
              aria-invalid="false"
              class="chakra-checkbox__input"
              id="field-«r7»"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-1dnp747"
            />
            <span
              class="chakra-checkbox__label css-6x44c9"
            >
              Makan Siang
            </span>
          </label>
          <label
            class="chakra-checkbox css-1577qb8"
          >
            <input
              aria-checked="false"
              aria-disabled="false"
              aria-invalid="false"
              class="chakra-checkbox__input"
              id="field-«r7»"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-1dnp747"
            />
            <span
              class="chakra-checkbox__label css-6x44c9"
            >
              Snack Sore
            </span>
          </label>
        </div>
      </div>
      <div
        class="chakra-form-control css-1n1vgb"
        role="group"
      >
        <label
          class="chakra-form__label css-g6pte"
          for="field-«r8»"
          id="field-«r8»-label"
        >
          Nominal Konsumsi
        </label>
        <div
          class="chakra-input__group css-1y0e7gb"
          data-group="true"
        >
          <div
            class="chakra-input__left-addon css-127xmqf"
          >
            Rp.
          </div>
          <input
            aria-readonly="true"
            class="chakra-input css-10wjqo2"
            id="field-«r8»"
            readonly=""
            value="0"
          />
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-rnrgf4"
      />
      <div
        class="css-9idvp9"
      >
        <button
          class="chakra-button css-1qtz3om"
          type="reset"
        >
          Batal
        </button>
        <button
          class="chakra-button css-ez23ye"
          disabled=""
          type="submit"
        >
          Simpan
        </button>
      </div>
    </form>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
